osquery/tables/system/python_packages.cpp

```diff
@@ -113,28 +136,119 @@ void genSiteDirectories(const std::string& site,
     r["directory"] = site;
     r["path"] = directory;
     r["pid_with_namespace"] = "0";
+    r["uid"] = BIGINT(user_id);
     results.push_back(r);
   }
 }
 
-void genWinPythonPackages(const std::string& keyGlob,
-                          QueryData& results,
-                          Logger& logger) {
+std::vector<std::string> listWinPythonPaths(const std::string& keyGlob) {
 #ifdef WIN32
+  std::vector<std::string> results;
+
   std::set<std::string> installPathKeys;
-  expandRegistryGlobs(keyGlob, installPathKeys);
+  auto status = expandRegistryGlobs(keyGlob, installPathKeys);
+  if (!status.ok()) {
+    return {};
+  }
+
   QueryData pythonInstallLocation;
   for (const auto& installKey : installPathKeys) {
     queryKey(installKey, pythonInstallLocation);
     for (const auto& p : pythonInstallLocation) {
       if (p.at("name") != "(Default)") {
         continue;
       }
-      genSiteDirectories(p.at("data"), results, logger);
+      results.push_back(p.at("data"));
     }
     pythonInstallLocation.clear();
+
+    return results;
   }
 #endif
+  return {};
+}
+
+std::vector<std::string> traverseVersions(const std::string& path) {
+  std::vector<std::string> versions;
+  if (!listDirectoriesInDirectory(path, versions, false).ok()) {
+    return {};
+  }
+
+  std::vector<std::string> all_paths;
+  for (const auto& version : versions) {
+    auto version_path = fs::path(version);
+    if (fs::is_symlink(symlink_status(version_path))) {
+      continue;
+    }
+
+    auto path = version + "/lib/python%/site-packages";
+    std::vector<std::string> sites;
+    resolveFilePattern(path, sites);
+    for (const auto& site : sites) {
+      all_paths.push_back(site);
+    }
+  }
+
+  return all_paths;
+}
+
+std::vector<std::map<std::string, UserPath>> getUserPathList(
+    const QueryContext& context) {
+  std::vector<std::map<std::string, UserPath>> paths_list;
+
+  auto user_list = usersFromContext(context);
+  for (const auto& user : user_list) {
+    if (user.count("uid") == 0 || user.count("directory") == 0) {
+      continue;
+    }
+
+    const auto& uid_as_string = user.at("uid");
+    auto uid_as_big_int = tryTo<int64_t>(uid_as_string, 10);
+    if (uid_as_big_int.isError()) {
+      LOG(ERROR) << "Invalid uid field returned: " << uid_as_string;
+      continue;
+    }
+    const auto& path = user.at("directory");
+
+    if (isPlatform(PlatformType::TYPE_OSX)) {
+      std::set<std::string> user_paths = kUserDirectoryPaths;
+
+      user_paths.insert(kDarwinUserDirectoryPaths.begin(),
+                        kDarwinUserDirectoryPaths.end());
```

Comment:
I actually noticed this should
```
if (!isPlatform(PlatformType::TYPE_WINDOWS)) {
      std::set<std::string> user_paths = kUserDirectoryPaths;

      if (isPlatform(PlatformType::TYPE_OSX)) {
          user_paths.insert(kDarwinUserDirectoryPaths.begin(),
                                         kDarwinUserDirectoryPaths.end());
      }
```
Otherwise we are going to miss `.pyenv` for linux